suricata/update/parsers.py

```diff
@@ -0,0 +1,244 @@
+            {'metavar': '<filename>',
+                'help': "Output YAML fragment for rule inclusion"}),
+        (("--url",),
+            {'metavar': '<url>', 'action': 'append', 'default': [],
+                'help': "URL to use instead of auto-generating one "
+                "(can be specified multiple times)"}),
+        (("--local",),
+            {'metavar': '<path>', 'action': 'append', 'default': [],
+                'help': "Local rule files or directories "
+                "(can be specified multiple times)"}),
+        (("--sid-msg-map",),
+            {'metavar': '<filename>',
+                'help': "Generate a sid-msg.map file"}),
+        (("--sid-msg-map-2",),
+            {'metavar': '<filename>',
+                'help': "Generate a v2 sid-msg.map file"}),
+
+        (("--disable-conf",),
+            {'metavar': '<filename>',
+                'help': "Filename of rule disable filters"}),
+        (("--enable-conf",),
+            {'metavar': '<filename>',
+                'help': "Filename of rule enable filters"}),
+        (("--modify-conf",),
+            {'metavar': '<filename>',
+                'help': "Filename of rule modification filters"}),
+        (("--drop-conf",),
+            {'metavar': '<filename>',
+                'help': "Filename of drop rule filters"}),
+
+        (("--ignore",),
+            {'metavar': '<pattern>', 'action': 'append', 'default': None,
+                'help': "Filenames to ignore "
+                "(can be specified multiple times; default: *deleted.rules)"}),
+        (("--no-ignore",),
+            {'action': 'store_true', 'default': False,
+                'help': "Disables the ignore option."}),
+        (("--threshold-in",),
+            {'metavar': '<filename>',
+                'help': "Filename of rule thresholding configuration"}),
+        (("--threshold-out",),
+            {'metavar': '<filename>',
+                'help': "Output of processed threshold configuration"}),
+        (("--dump-sample-configs",),
+            {'action': 'store_true', 'default': False,
+                'help': "Dump sample config files to current directory"}),
+        (("--etopen",),
+            {'action': 'store_true',
+                'help': "Use ET-Open rules (default)"}),
+        (("--reload-command",),
+            {'metavar': '<command>',
+                'help': "Command to run after update if modified"}),
+        (("--no-reload",),
+            {'action': 'store_true', 'default': False,
+                'help': "Disable reload"}),
+        (("-T", "--test-command"),
+            {'metavar': '<command>',
+                'help': "Command to test Suricata configuration"}),
+        (("--no-test",),
+            {'action': 'store_true', 'default': False,
+                'help': "Disable testing rules with Suricata"}),
+        (("--no-merge",),
+            {'action': 'store_true', 'default': False,
+                'help': "Do not merge the rules into a single file"})
+    ]
```

Comment:
I think we should take this data structure and others out of the functions. Functions can be compact that way. At present, the function is mostly about declaration.

Reply:
@shivan1b I didn't very well understand it. If we move out the data structure, then what should we keep inside the function?

Reply:
The logic. We can move this list out of the function and keep the operations on the list inside the function.

Reply:
Yes, I get that. We can do this with `parse_global` and `parse_update` functions. 
Does `parse_commands` looks fine to you?